### DISCUSSION TITLE
Using A17- Vue components for custom actions

### DISCUSSION THREAD

**Discussion by VizuaaLOG** [area17/twill#868] [Q&A]
Hey 👋 ,

I'm currently looking into Twill for a project I may be doing. It looks really solid, so great work! One of the things I'm currently struggling to work out is how to make the various `a17-` components available globally? For example, there will be parts of my project that require some areas to have custom actions, such as the ability to mark orders as completed, refund etc.

Looking through the source I can see there's a `@yield('publisherRows')` which allows me to add custom content to the publisher, or I can even do this via a side fieldset. I would ideally like to make use of the existing Twill Vue components to keep looks consistent, plus the `a17-multibutton` component looks great for doing things like changing the status of an order.

The issue I'm facing is Vue isn't displaying the button no matter where I put it, even inside the `publisherRows` blade section. I'm assuming this is because these generic button components, as well as other UI components, are not registered with Vue globally but rather on a per-component basis. E.g. the Publisher Vue component is registering the multi-button.

Dopes Twill provides a way to make use of these components globally or would I need to write an abstracted Vue component that acts as a bridge to import them?

It's also possible I'm completely missing something as I've been staring at this for awhile trying to figure out a way to do it 😅 .

Appreciate any support on this.

**Edit** I have now found that some components do work, like the button and dropdown. But I'm guessing to make use of the more complex components I would need a custom Vue component, since I would need methods on it etc.

**Comment by ptrckvzn** [comment]
Hi @VizuaaLOG, on the 2.x branch, I believe the form components and also custom components are loaded and should be available for the whole `#app` (which wrap the `publisherRows` section).

If you create custom components (for example in `resources/assets/js/components`, you would need to run `php artisan twill:build` in order to include them to the Twill assets, otherwise, you can see here in [`main-form.js`](https://github.com/area17/twill/blob/2.x/frontend/js/main-form.js#L124) where they are registered.

Let me know if that helps!
